sound/soc/sof/topology.c

```diff
@@ -286,11 +288,34 @@ static int sof_control_load_volume(struct snd_soc_component *scomp,
 	scontrol->num_channels = le32_to_cpu(mc->num_channels);
 
 	/* set cmd for mixer control */
-	if (mc->max > 1)
-		scontrol->cmd = SOF_CTRL_CMD_VOLUME;
-	else
+	if (mc->max <= 1) {
```

Comment:
can the max be zero? I think not so the test should be == maybe?

Reply:
@plbossart Updated. Yes, it should be 1 for switch, thanks. The interesting thing is that it is still valid if we set max = 0  to a volume kcontrol. But of cause it is muted in playback since the volume value is 0.